For original patch, please see:

From 0b28f1e3bae10def59e7fe8afac29ed5355a3f1e Mon Sep 17 00:00:00 2001
From: Andy Shaw <andy.shaw@qt.io>
Date: Tue, 21 Jan 2020 15:01:55 +0100
Subject: [PATCH] Fix QLocale::system() and uiLanguages() for the mobile platforms

On iOS, it was no longer using the Cocoa specific code needed to get
the locale and uiLanguages information so this functionality is brought
back as it was accidently lost.

In addition, this has a side-effect of fixing a problem with Android
versions below API 24 where it has no UiLanguages functionality so it
gets the information based on the system locale as a fallback.

Fixes: QTBUG-81307
Fixes: QTBUG-81357
Change-Id: I1709675b5bd5e9cedefb99eaec28279f20a347a4
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
---

diff --git a/src/corelib/tools/qlocale.cpp b/src/corelib/tools/qlocale.cpp
@@ -4463,6 +4463,8 @@
             for (const auto entry : qAsConst(uiLanguages))
                 locales.append(QLocale(entry));
         }
+        if (locales.isEmpty())
+            locales.append(systemLocale()->fallbackUiLocale());
     } else
 #endif
     {
